Localhost/Mac/System/telemetry/installer/mac/functions.bash

```diff
@@ -0,0 +1,204 @@
+    write_log "INFO" "Getting the selected products ..."
+
+    local telemetries=$(jq -c '.configuration.subtypes[0].datasources[0].telemetries[]' $app_json)
+    if [[ "$telemetries" = null ]]; then
+        write_run "print_error \"installer.bash (1): .configuration.subtypes[0].datasources[0].telemetries[] was not found in application JSON\""
+        return 1
+    fi
+    if [[ -z "$telemetries" ]]; then
+        write_run "print_error \"installer.bash (1): .configuration.subtypes[0].datasources[0].telemetries[] is empty in application JSON\""
+        return 1
+    fi
+
+    local is_logs_option_selected=false
+    local is_metrics_option_selected=false
+    local index=0
+
+    while read -r telemetry; do
+        local type=$(echo "$telemetry" | jq -r '.type')
+        if [[ "$type" = null ]]; then
+            write_run "print_error \"installer.bash (1): '.configuration.subtypes[0].datasources[0].telemetries[$index].type' was not found in application JSON\""
+            return 1
+        fi
+        if [[ -z "$type" ]]; then
+            write_run "print_error \"installer.bash (1): '.configuration.subtypes[0].datasources[0].telemetries[$index].type' is empty in application JSON\""
+            return 1
+        fi
+
+        local params=$(echo -e "$telemetry" | jq -r '.params[]')
+        if [[ "$params" = null ]]; then
+            write_run "print_error \"installer.bash (1): '.configuration.subtypes[0].datasources[0].telemetries[$index].params[]' was not found in application JSON\""
+            return 1
+        fi
+
+        if [[ "$type" = "LOG_ANALYTICS" ]]; then
+            write_log "INFO" "is_logs_option_selected = true"
+            write_log "INFO" "logs_params = $params"
+
+            is_logs_option_selected=true
+            write_run "logs_params='$params'"
+        elif [[ "$type" = "METRICS" ]]; then
+            write_log "INFO" "is_metrics_option_selected = true"
+            write_log "INFO" "metrics_params = $params"
+
+            is_metrics_option_selected=true
+            write_run "metrics_params='$params'"
+        fi
+
+        let "index++"
+    done < <(echo -e "$telemetries")
+
+    write_run "is_logs_option_selected=$is_logs_option_selected"
+    write_run "is_metrics_option_selected=$is_metrics_option_selected"
+}
+
+# Creates Logz.io opt directory
+function create_logzio_opt_dir () {
+    write_log "INFO" "Creating Logz.io opt directory ..."
+
+    logzio_opt_dir="/opt/logzio-otel-collector"
+    mkdir -p $logzio_opt_dir
+    write_run "logzio_opt_dir=\"$logzio_opt_dir\""
+}
+
+# Gets OTEL collector binary
+# Output:
+#   otel_bin - The OTEL collector binary file path
+# Error:
+#   Exit Code 2
+function get_otel_collector_binary () {
+    write_log "INFO" "Getting OTEL collector binary ..."
+    curl -fsSL https://github.com/open-telemetry/opentelemetry-collector-releases/releases/download/v0.56.0/otelcol-contrib_0.56.0_darwin_amd64.tar.gz > $logzio_temp_dir/otelcol-contrib.tar.gz 2>$task_error_file
```

Comment:
please downgrade to version 0.55
There is a bug with network metrics on some of the machines